### PR TITLE
Verify invoice amount in lightning address withdrawal

### PR DESCRIPTION
**Pull request by benthecarman**
Not sure how to test, but looking at the `decodePaymentRequest` docs this looks correct.

### CODE REVIEW

**Comment by ekzyis** [comment]
> Not sure how to test

You would need to setup local lnd as explained [here](https://github.com/stackernews/stacker.news/blob/master/docs/local-lnd.md). But I can take a look at this today :)

File: api/resolvers/wallet.js [hunk 1]
```diff
@@ -313,6 +313,10 @@ export default {
         throw new Error('description hash does not match')
       }
 
+      if (decoded.num_msat !== milliamount) {
+        throw new Err('invoice has incorrect amount')
```
Comment (ekzyis) [inline api/resolvers/wallet.js:317]:
```suggestion
        throw new Error('invoice has incorrect amount')
```

**Review by ekzyis — CHANGES_REQUESTED** [review]

File: api/resolvers/wallet.js [hunk 2]
```diff
@@ -313,6 +313,9 @@ export default {
         throw new Error('description hash does not match')
       }
 
+      if (decoded.num_msat !== milliamount) {
```
Comment (ekzyis) [inline api/resolvers/wallet.js:316]:
```suggestion
      if (!decoded.mtokens || BigInt(decoded.mtokens) !== BigInt(milliamount)) {
```

**Review by ekzyis — CHANGES_REQUESTED** [review]
I've tested this by sending to another SN account via its LN address on my local instance.

I've noticed that `num_msat` is undefined. `decodePaymentRequest` returned this for a 1 sat invoice:

```
app  | {
app  |   chain_address: undefined,
app  |   cltv_delta: 80,
app  |   created_at: '2023-09-20T15:19:05.000Z',
app  |   description: '',
app  |   description_hash: '3642f0f74ffb44596b4be110fefa2ebfe5cedc209e8e376d9b41e1ad0938d3a5',
app  |   destination: '0229ad2b79601ca9831128a10398aa4fc8a0bcfb568bdddb5955c9259834baf7d8',
app  |   expires_at: '2023-09-20T15:20:05.000Z',
app  |   features: [
app  |     { bit: 9, is_known: true, is_required: false, type: 'tlv_onion' },
app  |     {
app  |       bit: 14,
app  |       is_known: true,
app  |       is_required: true,
app  |       type: 'payment_identifier'
app  |     },
app  |     {
app  |       bit: 17,
app  |       is_known: true,
app  |       is_required: false,
app  |       type: 'multipath_payments_v0'
app  |     }
app  |   ],
app  |   id: '4e7194285ea4955cc7aad374873a97f8775bfa6c141bae203727bd4adbadf6b1',
app  |   is_expired: false,
app  |   mtokens: '1000',
app  |   payment: '07a6d31bca2b6aeaeece0a2baedd75976f119d66c58e2aafef2c7ff0eab9ee14',
app  |   routes: [],
app  |   safe_tokens: 1,
app  |   tokens: 1
app  | }
```

I also looked at the [docs](https://www.npmjs.com/package/ln-service#decodepaymentrequest) and I see no `num_msat`. I think we need to use `BigInt(decoded.mtokens)` as we do in other parts of the code like [here](https://github.com/stackernews/stacker.news/blob/master/api/resolvers/wallet.js#L382).

_(While doing this, I realized we also need to document `LNWITH_URL` in `.env.sample`)_

**Comment by benthecarman** [comment]
Thanks @ekzyis! I was looking at the wrong docs I guess :sweat_smile: 

https://lightning.engineering/api-docs/api/lnd/lightning/decode-pay-req

**Review by ekzyis — APPROVED** [review]
If you want to have "verified stacker.news contributor" in your SN profile, you can add your SN nym to `contributors.txt` in this PR :)

**Comment by benthecarman** [comment]
Cleaned up commits + added to list!

**Comment by huumn** [comment]
I'll likely deploy this monday. 😉

**Pull request merged**